whisper/whisperv6/whisper.go

```diff
@@ -553,6 +554,24 @@ func (whisper *Whisper) Subscribe(f *Filter) (string, error) {
 	return s, err
 }
 
+// rebuildBloomFilter makes a new bloom filter from currently subscribed watchers
+// if new bloom filter is different from current bloom filter - peers will be notified
+func (whisper *Whisper) rebuildBloomFilter() {
+	if isFullNode(whisper.BloomFilter()) {
+		return
+	}
+	aggregate := make([]byte, bloomFilterSize)
+	whisper.filters.Each(func(filter *Filter) {
```

Comment:
The thing is, as of https://github.com/ethereum/go-ethereum/pull/16176 we no longer remove filters automatically. I expect this to be way less effective. @gluk256 we should make filter removal an option, so that people who want to keep those filters can keep them, and those wanting a more unforgiving approach can do so as well.